pandasai/__init__.py

```diff
@@ -703,6 +705,115 @@ def _retry_run_code(self, code: str, e: Exception, multiple: bool = False):
             self.callback.on_code(code)
         return code
 
+    def _execute_catching_errors(
+        self, code: str, environment: dict
+    ) -> Optional[Exception]:
+        """
+        Perform execution of the code directly.
+
+        Call `exec()` for the given `code`, catch any non-base exceptions.
+
+        Args:
+            code (str): Python code
+            environment (dict): Context for the `exec()`
+
+        Returns (Optional[Exception]): Any exception raised during execution.
+                                       `None` if executed without exceptions.
+        """
+        try:
+            exec(code, environment)
+        except Exception as exc:
+            self.log("Error of executing code", level=logging.WARNING)
+            self.log(f"{traceback.format_exc()}", level=logging.DEBUG)
+
+            return exc
+
+    def handle_error(
```

Comment:
IMO here we should definitely check whether what we are trying to import is indeed a whitelisted library, or it's just too dangerous!

Reply:
> we should definitely check whether what we are trying to import is indeed a whitelisted library
EXACTLY, had the same thought, i just needed to hear confirmation for this from you

Reply:
Sure, let's go for the allowing only whitelisted then! Thanks a lot!